common/lib/client/realtimechannel.js

```diff
@@ -505,12 +498,12 @@ var RealtimeChannel = (function() {
 		switch(this.state) {
 			case 'attaching':
 				var err = new ErrorInfo('Channel attach timed out', 90000, 408);
-				this.setState('detached', err);
-				this.failPendingMessages(err);
+				this.notifyState('suspended', err);
+				this.failPendingMessages(err); // RTP5f
 				break;
 			case 'detaching':
 				var err = new ErrorInfo('Channel detach timed out', 90000, 408);
-				this.setState('attached', err);
+				this.notifyState('attached', err);
```

Comment:
I vaguely recall discussing this, but wonder if we should modify the spec to state that going from detaching to attached always resends the `ATTACH` protocol message in case somehow, the `DETACH` arrived but the client never got the `DETACHED`.  By simply resending `ATTACH` for this edge case we rule out any possibility that the two are not kept in sync.  WDYT? cc @paddybyers